Sources/OAuth2Client/NXOAuth2Connection.m

```diff
@@ -418,7 +418,7 @@ - (void)connection:(NSURLConnection *)connection didReceiveResponse:(NSURLRespon
     if (self.statusCode == 401
         && client.accessToken.refreshToken != nil
         && authenticateHeader
-        && [authenticateHeader rangeOfString:@"expired_token"].location != NSNotFound) {
+        && ([authenticateHeader rangeOfString:@"invalid_token"].location != NSNotFound || [authenticateHeader rangeOfString:@"invalid_request"].location != NSNotFound)) {
```

Comment:
Maybe, but it has permanently fixed our crashes. We have had not issues since implementing this. I understand it may not be correct generally, but we had to get this working. Thanks for looking at it!!